src/lib/withGestures.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import Hammer from 'hammerjs'
+
+const shouldListenToSwipe = handlers => Object.keys(handlers).find(h => /^swipe/.test(h)) !== undefined
+const shouldListenToTap = handlers => Object.keys(handlers).find(h => /^tap/.test(h)) !== undefined
+
+const withGestures = (eventHandlers) => {
+  return WrappedComponent => {
+    return class WithGesturesComponent extends Component {
+      componentDidMount () {
+        this.hammer = new Hammer(ReactDOM.findDOMNode(this))
```

Comment:
Swipe up and down don't work at the moment because Hammer [disables them by default](http://hammerjs.github.io/getting-started/).
You can add a line such as `this.hammer.get('swipe').set({ direction: Hammer.DIRECTION_VERTICAL })` and it will work. I also suggest we disable all recognizers except tap and swipe, since we don't use them at the moment.